mining/mining.go

```diff
@@ -620,14 +627,38 @@ mempoolLoop:
 		switch {
 		// If segregated witness has not been activated yet, then we
 		// shouldn't include any witness transactions in the block.
-		case tx.MsgTx().HasWitness() && !includeWitness:
+		case tx.MsgTx().HasWitness() && !segwitActive:
```

Comment:
The `HasWitness` call has been cached in a later commit, however I went ahead and swapped the order as well.